vector/src/main/scala/geotrellis/vector/Point.scala

```diff
@@ -321,4 +322,16 @@ case class Point(jtsGeom: jts.Point) extends Geometry
    */
   def within(g: Geometry): Boolean =
     jtsGeom.within(g.jtsGeom)
+
+  def isInRing(l: Line): Boolean =
+    CGAlgorithms.isPointInRing(jtsGeom.getCoordinate, l.jtsGeom.getCoordinates)
```

Comment:
I'd be curious what a benchmark between this and another, more general version (Polygon.contains). If there's no significant performance improvement, then I'm not sure we need to implement this. If there is a significant performance improvement, we'll have to think about when the user would want to use this function (and the others), and try to guide that usage through the API, the type system and/or documentation.

You can benchmark this using the [geotrellis-benchmark](https://github.com/geotrellis/geotrellis-benchmark) project.